### PR TITLE
Refs [TASK] [28528] Add load more trending track

### PR DESCRIPTION
**Pull request by phong016688**
## Related Tickets
- [#28528: {Add loadmore track}](https://edu-redmine.sun-asterisk.vn/issues/28528)

## WHAT
- Thêm loadmore cho list nhac

## Evidence (Screenshot or Video)
image

## Review Checklist
First, self review then send pull request to trainer
Category | View Point | Description | Expected Reviewer Answer | Self review | Trainer reivew (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the source code have end-line errors? | | NO |<li>- [ ] no</li>|<li>- [ ] no</li>  
Conventions | No hard coding, use constants/configuration values | | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>

### CODE REVIEW

File: app/src/main/java/com/example/sunmusic/screen/trending/TrendingAdapter.kt [hunk 1]
```diff
@@ -41,18 +48,42 @@ class TrendingAdapter : RecyclerView.Adapter<BaseViewHolder<TrendingItem>>() {
...
-        listItem.addAll(items)
-        listItem.sortBy { it.type }
-        if (listItem.size > 0) {
-            notifyItemRangeInserted(oldSizeListItem, items.size)
-        } else {
-            listItem.clear()
-            notifyDataSetChanged()
+    fun addTitleTrackItem() {
+        listItem.add(TrendingItem.TitleTrack)
+        //notifyItemInserted(itemCount)
```
Comment (duynn-2359) [inline app/src/main/java/com/example/sunmusic/screen/trending/TrendingAdapter.kt]: không dùng thì remove

File: app/src/main/java/com/example/sunmusic/screen/trending/TrendingFragment.kt [hunk 2]
```diff
@@ -31,28 +36,50 @@ class TrendingFragment : Fragment(R.layout.fragment_trending), TrendingContract.
         view?.trendingRecyclerView?.apply {
             adapter = this@TrendingFragment.adapter
             hasFixedSize()
+            val linearLayoutManager = layoutManager ?: return@apply
+            val onLoadMore = {
+                this@TrendingFragment.adapter.startLoadMore()
+                presenter.loadMoreTopTracks(offset = currentOffsetTrack)
+                currentOffsetTrack += 5
```
Comment (duynn-2359) [inline app/src/main/java/com/example/sunmusic/screen/trending/TrendingFragment.kt]: hard code

File: app/src/main/java/com/example/sunmusic/screen/trending/TrendingAdapter.kt [hunk 3]
```diff
@@ -41,18 +48,42 @@ class TrendingAdapter : RecyclerView.Adapter<BaseViewHolder<TrendingItem>>() {
...
+        notifyDataSetChanged()
+    }
+
+    fun addTopAlbumItem(topAlbumItem: TrendingItem.AlbumsItem) {
+        listItem.add(0, topAlbumItem)
+        notifyItemInserted(1)
```
Comment (duynn-2359) [inline app/src/main/java/com/example/sunmusic/screen/trending/TrendingAdapter.kt]: hard code

File: app/src/main/java/com/example/sunmusic/screen/trending/TrendingAdapter.kt [hunk 4]
```diff
@@ -41,18 +48,42 @@ class TrendingAdapter : RecyclerView.Adapter<BaseViewHolder<TrendingItem>>() {
...
+    }
+
+    fun addListTrackItem(trackItems: List<TrendingItem.TrackItem>) {
+        listItem.addAll(trackItems)
+        notifyItemRangeInserted(itemCount + 1, trackItems.size)
```
Comment (duynn-2359) [inline app/src/main/java/com/example/sunmusic/screen/trending/TrendingAdapter.kt]: as above

File: app/src/main/java/com/example/sunmusic/screen/trending/TrendingAdapter.kt (hunk 1 above)
Comment (tuanlvt-1117) [inline app/src/main/java/com/example/sunmusic/screen/trending/TrendingAdapter.kt]: remove

File: app/src/main/java/com/example/sunmusic/screen/trending/TrendingFragment.kt (hunk 2 above)
Comment (tuanlvt-1117) [inline app/src/main/java/com/example/sunmusic/screen/trending/TrendingFragment.kt]: hard code (5)

File: app/src/main/java/com/example/sunmusic/utils/Constant.kt [hunk 5]
```diff
@@ -9,6 +9,11 @@ object Constant {
     const val READ_TIME_OUT = 10000
     const val DEFAULT_TOP_ALBUM_COUNT = 3
     const val DEFAULT_TOP_TRACK_COUNT = 10
+    const val LOAD_MORE_COUNT = 5
+    const val DEFAULT_OFFSET = 0
+    const val FIRST_POSITION_NOTIFY = 1
+    const val FIRST_INDEX_LIST = 0
+    const val FIRST_ODER_TRACK = 1
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/example/sunmusic/utils/Constant.kt]: value trùng. e đặt lại tên để dùng chung nhé

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**